prime-router/src/test/kotlin/history/SubmissionHistoryTests.kt

```diff
@@ -1225,4 +1206,118 @@ class SubmissionHistoryTests {
+            "org",
+            "client",
+            "full-elr",
+            "externalName",
+            null,
+            null,
+            5,
+            7,
+            false
+        )
+
+        val testEnrich = DetailedSubmissionHistory(
+            1,
+            TaskAction.receive,
+            OffsetDateTime.now(),
+            HttpStatus.OK.value(),
+            mutableListOf(inputReport)
+        )
+
+        val logs = listOf(
+            DetailedActionLog(
+                ActionLogScope.item,
+                UUID.randomUUID(),
+                1,
+                null,
+                ActionLogLevel.error,
+                InvalidEquipmentMessage("")
+            ),
+            DetailedActionLog(
+                ActionLogScope.translation,
+                UUID.randomUUID(),
+                2,
+                null,
+                ActionLogLevel.filter,
+                ReportStreamFilterResult(
+                    "recvOrg1.recvSvc1",
+                    5,
+                    "matches",
+                    listOf(
+                        "ordering_facility_county",
+                        "QUALITY_PASS"
+                    ),
+                    "802798",
+                    ReportStreamFilterType.QUALITY_FILTER
+                )
+            ),
+            DetailedActionLog(
+                ActionLogScope.translation,
+                UUID.randomUUID(),
+                2,
+                null,
+                ActionLogLevel.filter,
+                ReportStreamFilterResult(
+                    "recvOrg2.recvSvc2",
+                    5,
+                    "matches",
+                    listOf(
+                        "ordering_facility_county",
+                        "QUALITY_PASS"
+                    ),
+                    "802798",
+                    ReportStreamFilterType.QUALITY_FILTER
+                )
+            ),
+            DetailedActionLog(
+                ActionLogScope.translation,
+                UUID.randomUUID(),
+                2,
+                null,
+                ActionLogLevel.filter,
+                ReportStreamFilterResult(
+                    "recvOrg1.recvSvc1",
+                    5,
+                    "matches",
+                    listOf(
+                        "ordering_facility_county",
+                        "QUALITY_PASS"
+                    ),
+                    "802798",
+                    ReportStreamFilterType.QUALITY_FILTER
+                )
+            ),
+        )
+
+        assertThat(testEnrich.destinations.count()).isEqualTo(0)
+        testEnrich.enrichWithDescendants(
+            listOf(
+                DetailedSubmissionHistory(
+                    2,
+                    TaskAction.route,
+                    OffsetDateTime.now(),
+                    HttpStatus.OK.value(),
+                    null,
+                    logs
+                ),
+            )
+        )
+
+        testEnrich.run {
```

Comment:
This feels you are only testing happy day scenarios. Are there any error states to test? The code coverage shows you are missing some branches of the code. Consider refactoring some of the code out to its own functions if that makes it easier for testing.

Reply:
Refactored code into separate functions to make it more readable and consistent with the other enrich functions.  When I debug the unit test, it shows it goes through all the branches of the function, but I also see the coverage is missing some of that. So not sure what's happening there...